app/web/features/communities/EditCommunityInfoPage.tsx

```diff
@@ -106,14 +111,27 @@ export default function EditCommunityPage() {
     }
   );
 
+  const router = useRouter();
+
   return (
-    <CommunityBase>
+    <CommunityBase communityId={communityId}>
       {({ community }) => {
-        return community.mainPage?.canEdit ? (
+        if (!community.mainPage?.canEdit) {
+          return (
+            <Redirect
+              to={routeToCommunity(
+                community.communityId,
+                community.slug,
+                "info"
+              )}
+            />
+          );
+        }
+        return (
           <>
             <HtmlMeta title={EDIT_LOCAL_INFO} />
-            <PageHeader page={community.mainPage} />
-            <CommunityPageSubHeader community={community} defaultTab="info" />
+            <PageHeader page={community.mainPage!} />
```

Comment:
It's because you flipped the boolean check - before if `community.mainPage?.canEdit` is true, then `mainPage` must exist so the TypeScript type narrowing correctly inferred it can't be null.

The opposite check doesn't give that same guarantee - i.e. if `!community.mainPage?.canEdit` is false, it could be falsey because `mainPage` is undefined or `canEdit` is false, so we'd need to do another check on `mainPage` to see if it's really not null.

Couldn't we keep the old logic? Seems like it's doing the same thing as before?